miasm2/ir/ir.py

```diff
@@ -173,21 +173,21 @@ def merge_multi_affect(self, affect_list):
                 continue
 
             # Find collision
-            e_colision = reduce(
-                lambda x, y: x + y, [e.get_modified_slice() for e in expr_list])
-
+            e_colision = reduce(lambda x, y: x.union(y),
+                                (e.get_modified_slice() for e in expr_list),
+                                set())
             # Sort interval collision
-            known_intervals = sorted([(x[1], x[2]) for x in set(e_colision)])
+            known_intervals = sorted([(x[1], x[2]) for x in e_colision])
 
             # Fill with missing data
-            missing_i = get_missing_interval(known_intervals, 0, e.src.size)
+            missing_i = get_missing_interval(known_intervals, 0, dst.size)
 
-            rest = [(m2_expr.ExprSlice(dst, r[0], r[1]), r[0], r[1])
-                    for r in missing_i]
+            remaining = [(m2_expr.ExprSlice(dst, *interval),
```

Comment:
Unless I'm wrong you do not use `remaining` after line 199 where a generator would be enough and use less memory. Just replace `[...]` by `(...)`:

```Python
remaining = ((m2_expr.ExprSlice(dst, *interval), interval[0], interval[1]) for interval in missing_i)